packages/dev/core/src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -64,15 +69,63 @@ export abstract class BaseCameraPointersInput implements ICameraInput<Camera> {
         this._shiftKey = false;
         this._buttonsPressed = 0;
 
-        this._pointerInput = (p) => {
-            const evt = <IPointerEvent>p.event;
+        this._connectedObserver = this.camera._deviceSourceManager!.onDeviceConnectedObservable.add((deviceSource) => {
+            if (deviceSource.deviceType === DeviceType.Mouse) {
+                this._mouseObserver = deviceSource.onInputChangedObservable.add((eventData) => {
+                    if (!("deltaY" in eventData)) {
+                        let type = PointerEventTypes.POINTERMOVE;
+
+                        if (eventData.inputIndex !== PointerInput.Move) {
+                            type = deviceSource.getInput(eventData.inputIndex) === 1 ? PointerEventTypes.POINTERDOWN : PointerEventTypes.POINTERUP;
+                        }
+
+                        this._pointerInput(eventData, type);
+
+                        if (Date.now() - this._previousStartingPointerTime < InputManager.DoubleClickDelay && !this._doubleClickOccured && type === PointerEventTypes.POINTERUP) {
+                            this._doubleClickOccured = true;
+                            type = PointerEventTypes.POINTERDOUBLETAP;
+                            this._pointerInput(eventData, type);
+                        } else {
+                            this._doubleClickOccured = false;
+                        }
+
+                        if (type === PointerEventTypes.POINTERUP || type === PointerEventTypes.POINTERDOUBLETAP) {
+                            this._previousStartingPointerTime = Date.now();
+                        }
+                    }
+                });
+            } else if (deviceSource.deviceType === DeviceType.Touch) {
+                this._touchObservers[deviceSource.deviceSlot] = deviceSource.onInputChangedObservable.add((eventData) => {
+                    let type = PointerEventTypes.POINTERMOVE;
+
+                    if (eventData.inputIndex !== PointerInput.Move) {
+                        type = deviceSource.getInput(eventData.inputIndex) === 1 ? PointerEventTypes.POINTERDOWN : PointerEventTypes.POINTERUP;
+                    }
+
+                    this._pointerInput(eventData, type);
+                });
+            }
+        });
+
+        this._disconnectedObserver = this.camera._deviceSourceManager!.onDeviceDisconnectedObservable.add((deviceSource) => {
+            if (deviceSource.deviceType === DeviceType.Mouse) {
+                deviceSource.onInputChangedObservable.remove(this._mouseObserver);
+                this._mouseObserver = null;
+            } else if (deviceSource.deviceType === DeviceType.Touch) {
+                deviceSource.onInputChangedObservable.remove(this._touchObservers[deviceSource.deviceSlot]);
+                this._touchObservers[deviceSource.deviceSlot] = null;
+            }
+        });
+
+        this._pointerInput = (p, t) => {
+            const evt = p;

```

Comment:
I'm not sure what this is. Why not this?

```suggestion
        this._pointerInput = (eventData, type) => {
```